railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -265,41 +265,42 @@ class AppGenerator < AppBase
       class_option :minimal, type: :boolean, desc: "Preconfigure a minimal rails app"
       class_option :javascript, type: :string, aliases: ["-j", "--js"], default: "importmap", desc: "Choose JavaScript approach [options: importmap (default), webpack, esbuild, rollup]"
       class_option :css, type: :string, aliases: "-c", desc: "Choose CSS processor [options: tailwind, bootstrap, bulma, postcss, sass... check https://github.com/rails/cssbundling-rails]"
-      class_option :skip_bundle, type: :boolean, aliases: "-B", default: false, desc: "Don't run bundle install"
-      class_option :skip_decrypted_diffs, type: :boolean, default: false, desc: "Don't configure git to show decrypted diffs of encrypted credentials"
+      class_option :skip_bundle, type: :boolean, aliases: "-B", default: nil, desc: "Don't run bundle install"
+      class_option :skip_decrypted_diffs, type: :boolean, default: nil, desc: "Don't configure git to show decrypted diffs of encrypted credentials"
 
       def initialize(*args)
         super
 
+        imply_options({
+          **OPTION_IMPLICATIONS,
+          (:minimal) => [
+            :skip_action_cable,
+            :skip_action_mailbox,
+            :skip_action_mailer,
+            :skip_action_text,
+            :skip_active_job,
+            :skip_active_storage,
+            :skip_bootsnap,
+            :skip_dev_gems,
+            :skip_hotwire,
+            :skip_javascript,
+            :skip_jbuilder,
+            :skip_system_test,
+          ],
+          (:api) => [
+            :skip_asset_pipeline,
+            :skip_javascript,
+          ],
```

Comment:
Should not this be in a constant as well?

Reply:
Not right now, at least.  The constant is shared between `AppGenerator` and `PluginGenerator`.  `:minimal` and `:api` are only for `AppGenerator`.

Reply:
Right, but inside `AppGenerator` they are constant, so why not extract that constant here like:

```
OPTION_IMPLICATIONS = [
  **AppBase::OPTION_IMPLICATIONS,
  # everything else
]
```

Reply:
Ah, I misunderstood.  I can do that.  Though that will move `{ :minimal => [...] }` far away from `meta_options: [:minimal]`.  (The option implications hash is the first argument of [`imply_options`](https://github.com/jonathanhefner/rails/blob/20b65938b13f5f368db6e9629df1ffdb9fa95f5b/railties/lib/rails/generators/app_base.rb#L178).  `:meta_options` is a separate kwarg of  `imply_options`.)

Reply:
Good point. Let's keep this way them.